### PR TITLE
container item itemgroup

### PR DESCRIPTION
**Pull request by KorGgenT**
#### Summary

```SUMMARY: Features "Allow itemgroups to have a container-item"```
#### Purpose of change
This allows you to define an itemgroup in such a way that the items from the item group spawn in a defined container. See the example tools_toolbox itemgroup.

#### Describe the solution
add `container-item` to itemgroup json, and put the spawned item(s) into the container at the end of the spawning function. I also noticed that this stuff really ought to be refactored because it's ancient and a bit hard to follow.

#### Testing
changed zombie deathdrops to the new tool_toolbox itemgroup and killed a zombie, it spawned with a toolbox that had some things in it.

#### Additional context
I have created a new item called toolbox_empty that shall be the replacement of the old style toolbox; all it is is aan empty container you can hold.

### CODE REVIEW

**Comment by jbytheway** [comment]
How have you handled the case of the items not fitting in the container.

**Comment by KorGgenT** [comment]
they just don't spawn

**Comment by DialoMalison** [comment]
Small nit, toolbox_empty seems to me to be a bit of a misnomer. The box won't be empty once someone puts something in it. Perhaps rename it to toolbox_metal. This would open the door for someone to create toolbox_wooden and toolbox_leather items later on.

**Comment by KorGgenT** [comment]
it's just an id, and `toolbox` was taken

**Comment by DialoMalison** [comment]
OK, no worries then.

**Comment by EvgenijM86** [comment]
Empty toolbox is not big enough to hold all of the tools that it previously did, so it is not a replacement.
